projects/elements/src/lib/lazy-elements/lazy-elements-loader.service.ts

```diff
@@ -190,26 +178,18 @@ export class LazyElementsLoaderService {
     return url.replace(/https?:\/\//, '');
   }
 
-  private isPromise<T>(obj: T | Promise<T>): obj is Promise<T> {
-    return typeof (obj as any)?.then === 'function';
-  }
-
   private handleHook(hook: Hook, tag: string): Promise<void> {
     try {
-      const result = hook(tag);
-      if (this.isPromise(result)) {
-        return result;
-      } else {
-        return Promise.resolve();
-      }
+      return Promise.resolve(hook(tag));
     } catch (err) {
       return Promise.reject(err);
     }
   }
 
-  private resolveImportMap(url: string) {
+  private async resolveImportMap(url: string) {
     const System = (window as any).System;
     if (System) {
+      await System.prepareImport();
```

Comment:
This is the fix for the main issue raised in this PR.

Reply:
I am not really sure if we should introduce `async / await` syntax as it does not play well with Angular Zone.js AFAIK currently?

Reply:
Let me research a bit more and understand the implications to make sure we are not introducing a bug with this change.

Reply:
@tomastrajan I did some more research and tests and this is the conclusion I got: 
 - Zone.js does not play well with Promises or Async/Await (it's not `Async`/`Await` syntax issue, it's that it can't hook to promises completion as it does with synchronous code)
 - So to work that around we need to use `ChangeDetectorRef` to mark for change as we are already doing in the directive, so the current code actually was already done to support `async`/`await` and/or `promises`
 - I did a couple of tests and the use of await here or marking the method as aync does not add any challenge or different behaviour.